scripts/wawCommons.py

```diff
@@ -254,63 +254,74 @@ def absoluteFilePaths(directory, patterns=['*']):
                 yield os.path.abspath(os.path.join(dirpath, f))
 
 def _fileMatchesPatterns(filename, patterns):
-    """Helper function which checks if file matches one the patterns."""
+    """Helper function which checks if file matches one of the patterns."""
     for pattern in patterns:
         if fnmatch.fnmatchcase(filename, pattern):
             return True
     return False
 
-def getWorkspaceId(config, workspacesUrl, version, username, password):
+def getWorkspaces(workspacesUrl, version, username, password):
+    """
+    Returns a list of all workspaces that match given criteria.
+    """
 
-    workspaceId = getOptionalParameter(config, 'conversation_workspace_id')
+    # get all workspaces
+    requestUrl = workspacesUrl + '?version=' + version
+    logger.info("request url: %s", requestUrl)
+    response = requests.get(requestUrl, auth=(username, password))
+    responseJson = response.json()
+    logger.debug("response: %s", responseJson)
+    if not errorsInResponse(responseJson):
+        logger.info('Workspaces successfully retrieved.')
+    else:
+        logger.error('Cannot retrieve workspaces.')
+        sys.exit(1)
 
-    if not workspaceId:
-        workspaceId = ""
+    if 'workspaces' in responseJson:
+        return responseJson['workspaces']
+    else:
+        logger.error('No workspaces key in the response')
+        sys.exit(1)
+
+def filterWorkspaces(config, workspaces):
+
+    matchingWorkspaces = []
+
+    matchByName = getOptionalParameter(config, 'conversation_workspace_match_by_name')
+    # workspace is matched by name (or pattern if defined)
+    if matchByName in ["true", "True"]:
+        logger.info("workspace is matched by 'conversation_workspace_name' (or by 'conversation_workspace_name_pattern' if defined)")
+        workspaceNamePattern = getOptionalParameter(config, 'conversation_workspace_name_pattern')
+        if workspaceNamePattern is None:
+            workspaceNamePattern = getOptionalParameter(config, 'conversation_workspace_name')
```

Comment:
We should at least print warning when user specifies both `conversation_workspace_name_pattern` and `conversation_workspace_name`. Or you can use `getParametersCombination` function written by @ondrejtomas .

Reply:
You can use one config file for all scripts and it can happen that you have workspace name specified for uploading workspace and at the same time you have pattern there for deleting all e.g. test workspaces..